report/sections/Chapter 3/Specification.tex

```diff
@@ -0,0 +1,71 @@
+For this project we will only focus on creating a platform for the functional programming course, though it could be modified to support other languages we will not do this during this project. 
+
+\section*{Target users}
+The system will have two main users, the lecturer that will conduct the course and create the exercises, and the university students who will try to solve the exercises.
+
+
+\section*{Constraints of the project}
+Given this project is developed as a semester project at \aau there will be some constraints that we must adhere to. 
+One of the constraints is the time constraint, we have single semester to develop the system. 
+We also had to fulfill the study regulations. The study regulations requires that we create an internet based system, that is scalable. 
+ 
+\section*{Use Cases}
+The following section will present a number of use cases. 
+A description of the use case as well as the supporting features are then presented. 
+This is done in order to allow us to prioritize the features needed later on.
+
+These use cases will present the expected features that are meant to support the main feature
+\subsection*{Use case 0: Accounts}
+To keep track of what exercises each user has solved, and which permissions they have, a user should be able to log in to the platform, and be registered as a student or a lecturer.
+
+\subsection*{Use case 1: Solve exercises}
+A user should be able to choose an exercise session and a specific exercise. The exercise description is then retrieved and shown to the user.
+As the user completes the exercise, they should also be able to submit the exercise, and verify that their submitted solutions can compile and fulfills the requirements of the exercise. If it does not compile or fails the tests, it should provide the user with an error describing the problem. 
+
+\subsection*{Use case 2: Create exercises}
+A lecturer should be able to create and edit syllabi as well as the belonging exercise sessions and their included exercises. 
+For each given exercise the user should be able to specify an exercise description and tests that must be passed, the tests should be performed on the submitted code. Furthermore the user should be able to specify template code that will already be entered when the students opens the exercise.
+
+\subsection*{Use case 3: Show statistics over exercises}
+A lecturer should be able to view statistics about exercises, for example how many attempts were needed by users to complete it or the time taken, in order to determine if the difficulty is sufficient. To do This the system needs to log the information as the users attempt to complete the exercises.
```

Comment:
```suggestion
A lecturer should be able to view statistics about exercises, for example how many attempts were needed by users to complete it or the time taken, in order to determine if the difficulty is sufficient. To do This the system needs to log the information as the users attempt to complete the exercises.
```

We didn't actually do this. Should we change the use case to match? Or refine in MOSCOW?

Reply:
That is, we didn't prioritize this feature. The lecturer gets an overview of which students (and how many) successfully (and not) solved exercises.